drivers/bluetooth/hci/userchan.c

```diff
@@ -93,6 +93,53 @@ static struct net_buf *get_rx(const uint8_t *buf)
 	return NULL;
 }
 
+/**
+ * @brief Decode the length of an HCI H4 packet
+ * @details Decodes packet length according to Bluetooth spec v5 Vol 4 Part E
+ * @param packet_type	Type of packet according to Bluetooth spec v5.2 Vol 4, Part A Table 2.1
+ * @param packet_buff	Pointer to packet buffer
+ */
+static uint16_t decode_hci_packet_length(uint8_t packet_type, uint8_t *packet_buff)
+{
+	uint16_t payload_size = 0;
+	uint8_t header_size = 0;
+	const uint8_t packet_type_len = 1;
+
+	switch (packet_type) {
+	/* Bluetooth spec 5.2 Vol 4, Part E - 5.4.1 HCI Command Packet */
+	case H4_CMD:
+		payload_size = packet_buff[3];
+		header_size = 3;
+		break;
+	/* Bluetooth spec 5.2 Vol 4, Part E - 5.4.2 ACL Data Packets */
+	case H4_ACL:
+		payload_size = packet_buff[4] << 8 | packet_buff[3];
+		header_size = 4;
```

Comment:
```suggestion
		header_size = BT_HCI_ACL_HDR_SIZE;
```
or
```suggestion
		header_size = sizeof(struct bt_hci_acl_hdr);
```

Reply:
Thanks for bringing this up. Just a remark...for the HCI H4 SCO, I did not find a definition, should I add it to https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/bluetooth/hci_types.h ? 

e.g., 

```c
#define BT_HCI_SCO_HDR_SIZE 3
```

Reply:
Yes, please add the `BT_HCI_SCO_HDR_SIZE` along side the actual header - Something similar to 
```c
struct bt_hci_acl_hdr {
	uint16_t handle;
	uint16_t len;
} __packed;
```